File: chess-AI/chessPiece.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   chessPiece.cpp
 * Author: Jakub
 * 
 * Created on 16. ledna 2016, 14:15
 */

#include "chessPiece.h"

chessPiece::chessPiece() {
    colour = COLOUR::NONE;
    type = STONES::EMPTY;
}

chessPiece::chessPiece(const chessPiece& orig) {
    this->type = orig.type;
    this->colour = orig.colour;
}

chessPiece::chessPiece(colour_t orig, stones_t stone) {
    colour = orig;
    type = stone;
}

ostream& operator<<(ostream& os, const chessPiece &piece) {
    switch (piece.type) {
        case STONES::BISHOP:
            os << "B";
            break;
        case STONES::EMPTY:
            os << " ";
            break;
        case STONES::KING:
            os << "G";
            break;
        case STONES::KNIGHT:
            os << "K";
            break;
        case STONES::PAWN:
            os << "P";
            break;
        case STONES::QUEEN:
            os << "Q";
            break;
        case STONES::ROOK:
            os << "R";
            break;
    }
}

chessPiece& chessPiece::operator =(const chessPiece& orig)
{
    this->colour = orig.colour;
    this->type = orig.type;
}

void chessPiece::set(colour_t orig, stones_t stone)
{
    colour = orig;
    type = stone;
}

colour_t chessPiece::getColour()
{
    return colour;
}
stones_t chessPiece::getType()
{
    return type;
}
chessPiece::~chessPiece() {
}

ostream& operator<< (ostream& os, colour_t colour)
{
    switch(colour)
    {
        case COLOUR::WHITE:
            os << "WHITE";
            break;
        case COLOUR::BLACK:
            os << "BLACK";
            break;
        default:
            break;
    }
}

File: chess-lib/chess-lib/chessPiece.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace chess_lib
{
    public enum STONES
    {
        EMPTY = 0,
        PAWN = 1,
		KNIGHT = 2,
		BISHOP = 3,
		ROOK = 4,
		QUEEN = 5,
		KING = 6
    }

    public enum COLOUR
    {
        NONE = 0,
        BLACK = 1,
		WHITE = 2
    }

    class chessPiece
    {
        public COLOUR colour { get; private set; }
        public STONES type { get; private set; }

        chessPiece(chessPiece orig = null)
        {
            if(orig == null)
            {
                colour = orig.colour;
                type = orig.type;
            }
            else
            {
                colour = COLOUR.NONE;
                type = STONES.EMPTY;
            }
        }
        chessPiece(COLOUR orig, STONES stone)
        {
            colour = orig;
            type = stone;
        }

        void set(chessPiece orig)
        {
            set(orig.colour, orig.type);
        }
        void set(COLOUR orig, STONES stone)
        {
            colour = orig;
            type = stone;
        }        
        void clear()
        {
            type = STONES.EMPTY;
            colour = COLOUR.NONE;
        }

        public override string ToString()
        {
            return type.MyToString();
        }
    }
}
